picasso/src/main/java/com/squareup/picasso3/ContactsPhotoRequestHandler.java

```diff
@@ -65,12 +65,17 @@
         && matcher.match(data.uri) != UriMatcher.NO_MATCH);
   }
 
-  @Override public Result load(Request request, int networkPolicy) throws IOException {
-    InputStream is = getInputStream(request);
-    if (is == null) {
-      return null;
+  @Override public void load(Request request, int networkPolicy, Callback callback) {
+    try {
+      InputStream is = getInputStream(request);
+      if (is == null) {
+        callback.onError(new IOException("no contact found"));
+      } else {
+        callback.onSuccess(new Result(Okio.source(is), DISK));
+      }
+    } catch (Exception e) {
+      callback.onError(e);
```

Comment:
if `callback.onError()` throws, this calls `callback.onError()`. We need to do something separate to call the callback only 1x.

We have a `signalledCallback` thing in OkHttp when we have to solve this
https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/RealCall.java#L145

Reply:
i'm not sure the internal implementation of this callback is supposed to throw?...

Reply:
probably shouldn't, but added just in case.